src/benchmark/file_async_io_read_micro_benchmark.cpp

```diff
@@ -0,0 +1,174 @@
+#include <numeric>
+#include <random>
+#include <aio.h>
+#include "micro_benchmark_basic_fixture.hpp"
+
+namespace hyrise {
+
+const auto MB = uint32_t{1'000'000};
+
+class FileAsyncIOMicroReadBenchmarkFixture : public MicroBenchmarkBasicFixture {
+ public:
+  uint64_t control_sum = uint64_t{0};
+  std::vector<uint32_t> numbers = std::vector<uint32_t>{};
+  uint32_t vector_element_count;
+
+  void SetUp(::benchmark::State& state) override {
+    // TODO(everybody): Make setup/teardown global per file size to improve benchmark speed
+    ssize_t BUFFER_SIZE_MB = state.range(0);
+
+    // each int32_t contains four bytes
+    vector_element_count = (BUFFER_SIZE_MB * MB) / sizeof(uint32_t);
+    numbers = std::vector<uint32_t>(vector_element_count);
+    for (size_t index = 0; index < vector_element_count; ++index) {
+      numbers[index] = std::rand() % UINT32_MAX;
+    }
+    control_sum = std::accumulate(numbers.begin(), numbers.end(), uint64_t{0});
+
+    int32_t fd;
+    if ((fd = creat("file.txt", O_WRONLY)) < 1) {
+      std::cout << "create error" << std::endl;
+    }
+    //Assert((fd = creat("file.txt", O_WRONLY)) < 1, "create error");
+    chmod("file.txt", S_IRWXU);  // enables owner to rwx file
+    //Assert(write(fd, std::data(numbers), BUFFER_SIZE_MB * MB != BUFFER_SIZE_MB * MB), "write error");
+    if (write(fd, std::data(numbers), BUFFER_SIZE_MB * MB) != BUFFER_SIZE_MB * MB) {
+      std::cout << "write error" << std::endl;
+    }
+
+    close(fd);
+  }
+
+  void TearDown(::benchmark::State& /*state*/) override {
+    // TODO(everybody): Error handling
+    std::remove("file.txt");
+  }
+
+  void aio_read_error_handling(aiocb* aiocb, uint32_t EXPECTED_BYTES){
+    const auto err = aio_error(aiocb);
+    const auto ret = aio_return(aiocb);
+
+    if (err != 0) {
+      std::cout << "Error at aio_error(): " << std::strerror(errno) << std::endl;
+      close(aiocb->aio_fildes);
+      exit(2);
+    }
+
+    if (ret != static_cast<int32_t>(EXPECTED_BYTES)) {
+      std::cout << "Error at aio_return(). Got: " << ret << " Expected: " << EXPECTED_BYTES << std::endl;
+      close(aiocb->aio_fildes);
+      exit(2);
+    }
+  }
+
+ protected:
+};
+
+BENCHMARK_DEFINE_F(FileAsyncIOMicroReadBenchmarkFixture, AIO_READ_SEQUENTIAL)(benchmark::State& state) {  // open file
+  int32_t fd;
+  if ((fd = open("file.txt", O_RDONLY)) < 0) {
+    std::cout << "open error " << errno << std::endl;
+  }
+
+  const auto NUMBER_OF_BYTES = static_cast<uint32_t>(state.range(0) * MB);
+  const auto uint32_t_size = ssize_t{sizeof(uint32_t)};
+  const auto read_data_size = NUMBER_OF_BYTES / uint32_t_size;
+
+
+  for (auto _ : state) {
+    state.PauseTiming();
+
+    micro_benchmark_clear_disk_cache();
+    auto read_data = std::vector<uint32_t>{};
+    read_data.resize(read_data_size);
+
+    state.ResumeTiming();
+
+    struct aiocb aiocb;
+
+    memset(&aiocb, 0, sizeof(struct aiocb));
+    aiocb.aio_fildes = fd;
+    aiocb.aio_buf = std::data(read_data);
```

Comment:
do we actually do async io here?
here we read only once, I'd expect multiple reads being started here